src/main/java/hudson/plugins/accurev/delegates/AbstractModeDelegate.java

```diff
@@ -350,4 +399,91 @@ public void buildEnvVars(AbstractBuild<?, ?> build, Map<String, String> env) {
+     * version greater than 6 like 6.1.x or 7 then color parameter will streamStyle
+     * 
+     * @return
+     */
+    private String getStreamTypeParameter() {
+        String fullVersion = GetAccuRevVersion.getAccuRevVersion().trim();
+        String partialversion = fullVersion.substring(fullVersion.indexOf(" ") + 1);
+        String version = partialversion.substring(0, partialversion.indexOf(" "));
+        String[] versionSplits = version.split("\\.");
+        String type = ((Integer.parseInt(versionSplits[0]) < 6) || (Integer.parseInt(versionSplits[0]) == 6 && Integer
+                .parseInt(versionSplits[1]) < 1)) ? "style" : "streamStyle";
+        logger.info("Current AccuRev version " + fullVersion + " color type parameter " + type);
+        return type;
+    }
+
+    /**
+     * Get last transaction build from the jenkins for the currently running project
+     * @return
+     * @throws IOException
+     */
+    private String getLastBuildTransaction() throws IOException {
+        StringBuilder path = new StringBuilder(accurevEnv.get(JENKINS_HOME)).append("\\").append(JOBS).append("\\")
+                .append(accurevEnv.get(JOB_BASE_NAME));
+        File f = new File(path.toString(), ACCUREVLASTTRANSFILENAME);
+        if (!f.exists()) {
+            return null;
+        }
+        try (BufferedReader br = Files.newBufferedReader(f.toPath(), UTF_8)) {
+            return br.readLine();
+        }
+    }
+
+    /**
+     * Get list of new files to be added into the jenkins build from a given a transaction.
+     * @param lastTransaction
+     * @param stream
+     * @return
+     * @throws IOException
+     */
+
+    private String getFileRevisionsToBeIncluded(int lastTransaction, String stream) throws IOException {
+        List<AccurevTransaction> transactions = History.getTransactionsAfterLastTransaction(scm, server, accurevEnv,
+                accurevWorkingSpace, listener, launcher, stream, lastTransaction);
+     // collect all the files from the list of transactions and remove duplicates from the list of files.
+        List<String> fileRevisions = transactions.stream().filter(t -> t != null).map(t -> t.getAffectedPaths())
+                                     .flatMap(Collection<String>::stream).collect(Collectors.toList())
+                                     .parallelStream().distinct().collect(Collectors.toList());
+        return (!fileRevisions.isEmpty()) ? getPopulateFilePath(fileRevisions) : null;
+    }
+    /**
+     * Create a text file to keep the list of files to be populated.
+     * @param fileRevisions
+     * @return
+     */
+    private String getPopulateFilePath(List<String> fileRevisions) {
+        BufferedWriter bw = null;
+        File populateFile = null;
+        String filepath = null;
+        try {
+            StringBuilder path = new StringBuilder(accurevEnv.get(JENKINS_HOME)).append("\\").append(JOBS).append("\\")
```

Comment:
This won't work for folder plugin nor work on a Remote, please explain the purpose of the populate file?
This can definitely be implemented in a better way.